docs/gpu-support.md

```diff
@@ -0,0 +1,73 @@
+# Deploying with Accelerators in the Cloud HPC Toolkit
+
+## Supported modules
+
+* [vm-instance]
+* Slurm on GCP modules where applicable, both version 4 and version 5
+  * `schedmd-slurm-gcp-v5-*`
+  * `SchedMD-slurm-on-gcp-*`
+* PBS Pro modules (`pbspro-*`)
+
+## Accelerator definition automation
+
+The [vm-instance] and schedmd-slurm-gcp-v5 [node-group], [controller] and
+[login] modules support automation for defining the `guest_accelerator` config.
+If the user supplies any value for this setting, the automation will be bypassed.
+
+The automation is handled primary in the `gpu_definition.tf` files in these
+modules. This file assumes the existence of two input variables in the module:
+
+* `guest_accelerator`: A list of terraform objects with the attributes type and
+  count.
+* `machine_type`: Defines the machine type of the VM being created.
+
+`gpu_definition.tf` works by checking the `machine_type` to associate it with
+a GPU type and to extract the expected GPU count. The result will always
+define the maximum number of GPUs for the machine_type. If this is not desired,
+the `guest_accelerator` can be defined manually instead.
+
+This automation also only supports machine type `a2`. Machine type `n1` can also
+have guest accelerators attached, however the type and count
+cannot be determined automatically like with `a2`.
+
+[vm-instance]: https://github.com/GoogleCloudPlatform/hpc-toolkit/tree/main/modules/compute/vm-instance
+[node-group]: https://github.com/GoogleCloudPlatform/hpc-toolkit/tree/main/community/modules/compute/schedmd-slurm-gcp-v5-node-group
+[controller]: https://github.com/GoogleCloudPlatform/hpc-toolkit/tree/main/community/modules/scheduler/schedmd-slurm-gcp-v5-controller
+[login]: https://github.com/GoogleCloudPlatform/hpc-toolkit/tree/main/community/modules/scheduler/schedmd-slurm-gcp-v5-login
+
+## Troubleshooting and tips
+
+* To list accelerator types and availability by region, run
+  `gcloud compute accelerator-types list`. The information is also available in
+  the Google Cloud documentation [here](https://cloud.google.com/compute/docs/gpus/gpu-regions-zones).
+* Deployment time of VMs with guest accelerators can be longer than a simple VM.
+
+### Slurm on GCP
+
+* When deploying a Slurm cluster with GPUs, we highly recommend using the
+  modules based on Slurm on GCP version 5 (`schedmd-slurm-gcp-v5-*`). The
+  interface is more consistent with HPC Toolkit standards and more functionality
+  is available to support, debug and workaround any issues related to GPU
+  resources.
+* `SchedMD-slurm-on-gcp-v5-*` modules have a different interface for defining
+  attached accelerators, `gpu_type` and `gpu_count`. These must be set even if
+  the machine type implies GPUs.
+* Some GPUs will fail in Slurm on GCP v4 HPC Toolkit modules
+  (`SchedMD-slurm-on-gcp-*`) due to a timeout that cannot be customized.
+  * Slurm on GCP v5 HPC Toolkit modules (`schedmd-slurm-gcp-v5-*`) have a timeout
```

Comment:
I would make this a direct recommendation, and move this note to the troubleshooting and tips section.

Reply:
also, what is the name of the option? you should just recommend the setting=value pair (like 10 minutes).

Reply:
This is still under "Troubleshooting and tips", but in a subsection focused on Slurm on GCP specific issues.

I updated the text to be more clear and added an example updating `cloud_parameters`